src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexNode.cs

```diff
@@ -900,170 +1018,178 @@ RegexNode ExtractCommonPrefix()
                     return this;
                 }
 
-                // Process the first branch to get the maximum possible common string.
-                RegexNode? startingNode = FindBranchOneMultiStart(children[0]);
-                if (startingNode is null)
-                {
-                    return this;
-                }
-
-                RegexOptions startingNodeOptions = startingNode.Options;
-                string? originalStartingString = startingNode.Str;
-                ReadOnlySpan<char> startingSpan = startingNode.Type == One ? stackalloc char[1] { startingNode.Ch } : originalStartingString.AsSpan();
-                Debug.Assert(startingSpan.Length > 0);
-
-                // Now compare the rest of the branches against it.
-                for (int i = 1; i < children.Count; i++)
+                Span<char> scratchChar = new char[1];
```

Comment:
As written it can't just be changed, as later in the method it's assigned to `startingSpan`, which has an assigned heap lifetime and is thus incompatible.  I can hack it to have a local lifetime by first assigning it to `stackalloc char[0]`.